quinn/src/endpoint.rs

```diff
@@ -436,7 +443,12 @@ impl State {
                                         .send(ConnectionEvent::Proto(event));
                                 }
                                 Some(DatagramEvent::Response(t)) => {
-                                    self.outgoing.push_back(udp_transmit(t));
+                                    if self.transmit_queue_size.load(Ordering::Relaxed)
+                                        < MAX_TRANSMIT_QUEUE_SIZE
+                                    {
+                                        self.outgoing.push_back(udp_transmit(t));
+                                        self.transmit_queue_size.fetch_add(1, Ordering::Relaxed);
```

Comment:
How about we track the number of payload bytes (`t.contents.len()`) rather than the number of packets? This lets us reason in terms of approximate memory use directly, and ensures consistency regardless of segmentation offload.

Reply:
Great suggestion. I'll give it a try.

Reply:
I have changed to based on the bytes count of transmit.contents.len()